doc/stdlib/dates.rst

```diff
@@ -535,6 +535,31 @@ Periods
 
    Construct a ``Period`` type with the given ``v`` value. Input must be losslessly convertible to an ``Int64``\ .
 
+.. function:: CompoundPeriod(periods) -> CompoundPeriod
+
+   .. Docstring generated from Julia source
+
+   Construct a ``CompoundPeriod`` from a ``Vector`` of ``Period``\ s. The provided ``Period``\ s will be simplified using the following rules:
+
+   * All ``Period``\ s of the same type will be added together
```

Comment:
I ended up splitting the documentation between the `CompoundPeriod` type and the constructor. Let me know what you think.